### PR TITLE
Session 1 Attila

### PR DESCRIPTION
**Pull request by ClausPolanka**

### CODE REVIEW

File: src/__tests__/gilded_rose_spec.js [hunk 1]
```diff
@@ -1,11 +1,114 @@
...
+  describe("Shop", () => {
+    it("stores given items", () => {
+      const item = new Item("foo", 10, 40);
+      const shopItems = [item, item];
+      const shop = new Shop(shopItems);
+      expect(shop.items).toEqual(shopItems);
     });
 
+    describe("updateQuality", () => {
+      it("lowers sellIn and quality on each run", () => {
```
Comment (ClausPolanka) [inline src/__tests__/gilded_rose_spec.js:20]: What is the sellIn and the quality? By only reading this test, imo it's not clear to me

Comment (ClausPolanka) [reply src/__tests__/gilded_rose_spec.js:20]: What does 

> on each run

mean?

File: src/__tests__/gilded_rose_spec.js [hunk 2]
```diff
@@ -1,11 +1,114 @@
...
+        const shop = new Shop(items);
+        shop.updateQuality();
+
+        expect(shop.items).toEqual([
+          new Item("foo", 9, 39),
+          new Item("bar", 3, 19)
+        ]);
+      });
+
+      it("lowers quality by one if the sell date has not passed", () => {
```
Comment (ClausPolanka) [inline src/__tests__/gilded_rose_spec.js:31]: What is the quality here? Both numbers were lowered. I'd say this test is a duplicated one from above.

Comment (ClausPolanka) [reply src/__tests__/gilded_rose_spec.js:31]: What's the sell date. Last test there was the sellIn. Is there a difference?

File: src/__tests__/gilded_rose_spec.js [hunk 3]
```diff
@@ -1,11 +1,114 @@
...
+        const items = [new Item("foo", 2, 40)];
+        const shop = new Shop(items);
+        shop.updateQuality();
+        expect(shop.items).toEqual([new Item("foo", 1, 39)]);
+      });
+
+      it("lowers quality by one if the sell date has not passed", () => {
```
Comment (ClausPolanka) [inline src/__tests__/gilded_rose_spec.js:38]: Confusing test name. Some value gets lowered by two. And some value is even negative. Test name somehow gives focus on "one" and "not passed". Hard for the review to follow.

File: src/__tests__/gilded_rose_spec.js [hunk 4]
```diff
@@ -1,11 +1,114 @@
...
+        const items = [new Item("foo", 0, 40)];
+        const shop = new Shop(items);
+        shop.updateQuality();
+        expect(shop.items).toEqual([new Item("foo", -1, 38)]);
+      });
+
+      it("ensures quality of an item is never negative", () => {
```
Comment (ClausPolanka) [inline src/__tests__/gilded_rose_spec.js:45]: What is the quality in the setup of the test? There are two values. Both are 0. Which value is important.

File: src/__tests__/gilded_rose_spec.js [hunk 5]
```diff
@@ -1,11 +1,114 @@
...
+        const items = [new Item("foo", 0, 0)];
+        const shop = new Shop(items);
+        shop.updateQuality();
+        expect(shop.items[0].quality).toEqual(0);
+      });
+
+      describe("Aged Brie", () => {
+        it("increases in quality by 1 each run", () => {
```
Comment (ClausPolanka) [inline src/__tests__/gilded_rose_spec.js:53]: What does 
> each run

mean?

Comment (ClausPolanka) [reply src/__tests__/gilded_rose_spec.js:53]: What is the quality? There are two values. Both the same. On which value should the reviewer focus?

File: src/__tests__/gilded_rose_spec.js [hunk 6]
```diff
@@ -1,11 +1,114 @@
...
+        it("does not decrease in quality", () => {
+          const items = [new Item("Sulfuras, Hand of Ragnaros", 10, 80)];
+          const shop = new Shop(items);
+          shop.updateQuality();
+          expect(shop.items[0].quality).toEqual(80);
+        });
+      });
+
+      describe("Backstage passes to a TAFKAL80ETC concert", () => {
+        it("increases by 2 if there are 10 days or less", () => {
```
Comment (ClausPolanka) [inline src/__tests__/gilded_rose_spec.js:78]: "What" increases by two?

Comment (ClausPolanka) [reply src/__tests__/gilded_rose_spec.js:78]: Another test for the "less than 10" would have been nice.

File: src/__tests__/gilded_rose_spec.js [hunk 7]
```diff
@@ -1,11 +1,114 @@
...
+          const items = [
+            new Item("Backstage passes to a TAFKAL80ETC concert", 10, 20)
+          ];
+          const shop = new Shop(items);
+          shop.updateQuality();
+          expect(shop.items[0].quality).toEqual(22);
+        });
+
+        it("increases by 3 if there are 5 days or less", () => {
```
Comment (ClausPolanka) [inline src/__tests__/gilded_rose_spec.js:87]: Another test for "less than 5" would have been nice.

Comment (ClausPolanka) [reply src/__tests__/gilded_rose_spec.js:87]: "What" increases by 3?

File: src/__tests__/gilded_rose_spec.js [hunk 8]
```diff
@@ -1,11 +1,114 @@
...
+          const items = [
+            new Item("Backstage passes to a TAFKAL80ETC concert", 5, 20)
+          ];
+          const shop = new Shop(items);
+          shop.updateQuality();
+          expect(shop.items[0].quality).toEqual(23);
+        });
+
+        it("drops in quality to 0 when sellIn is less or equal to 0", () => {
```
Comment (ClausPolanka) [inline src/__tests__/gilded_rose_spec.js:96]: Another test for negative sellIn would have been nice.

**Comment by ClausPolanka** [comment]
At this point in time line coverage is 93%
